bids/layout/index.py

```diff
@@ -31,12 +31,12 @@ def _check_path_matches_patterns(path, patterns):
     """Check if the path matches at least one of the provided patterns. """
     if not patterns:
         return False
-    path = os.path.abspath(path)
+    path = path.absolute()
```

Comment:
```suggestion
    path = Path(path).absolute()
```
Allow strings coming in.

Reply:
`path` is always of type of `Path`: 
1. `index._check_path_matches_pattern` is called by `BIDSLayoutIndexer._validate_dir` and `BIDSLayoutIndexer._validate_file` only.
 2. `path` is passed to them from `BIDSLayoutIndexer._index_dir` and `BIDSLayoutIndexer._index_file` which use `pathlib.Path` objects.

And these are all private methods so we can assume that nothing else will call them.

So, there is no need to convert `path` to `Path`.